core/src/main/java/org/mskcc/oncotree/crosswalk/CrosswalkRepository.java

```diff
@@ -52,14 +55,19 @@ public MSKConcept queryCVS(String vocabularyId, String conceptId, String histolo
         try {
             ResponseEntity<MSKConcept> response = restTemplate.getForEntity(crosswalkURL, MSKConcept.class, vocabularyId, conceptId, histologyCode, siteCode);
             return response.getBody();
-        } catch (RestClientException e) {
-            logger.error("queryCVS() -- caught RestClientException: " + e);
-            throw new CrosswalkException("Exception while getting data from CVS Service with: \n " +
-                "URI: " + crosswalkURL + "\n" +
+        } catch (HttpClientErrorException e) {
```

Comment:
HttpClientErrorException is only one of the exceptions that might be thrown.  `restTemplate.getForEntity` throws RestClientException so you are missing exceptions like ResourceAccessException (e.g. if the URL is incorrect).  Also HttpClientErrorException is defined as "Exception thrown when an HTTP 4xx is received." so I don't think `else if (e.getStatusCode().is5xxServerError())` will ever be true.  I think this section needs to be re-written.  We can talk about alternatives if you want.